alkos/kernel/arch/x86_64/utils/serial/serial64.nasm

```diff
@@ -0,0 +1,163 @@
+; File: serial64.asm
+; Description: Serial port communication routines for 64-bit systems.
+;              Includes initialization, character output, and string output
+;              functions for COM1.
+; -----------------------------------------------------------------------------
+          bits 64
+
+; -----------------------------------------------------------------------------
+; Constants Definition
+; -----------------------------------------------------------------------------
+          COM1        equ 0x3F8    ; Base address for COM1
+          DLAB_BIT    equ 0x80     ; Divisor Latch Access Bit
+          BAUD_DIVISOR equ 3       ; Divisor for 38400 baud
+          EMPTY       equ 0x20     ; Line Status Register bit indicating empty
+          NEWLINE     equ 0x0A     ; Line Feed
+          SLASH_R     equ 0x0D     ; Carriage Return
+          NULL        equ 0        ; Null terminator for strings
+
+; -----------------------------------------------------------------------------
+; Section: Text (64-bit)
+; -----------------------------------------------------------------------------
+          section .text
+          global serial_init
+          global serial_putchar
+          global serial_puts
+
+; -----------------------------------------------------------------------------
+; Function: serial_init
+; Description:
+;     Initializes the serial port (COM1) for communication with the following
+;     settings:
+;         - Baud rate: 38400
+;         - Data bits: 8
+;         - Parity: None
+;         - Stop bits: 1
+; Parameters:
+;     None
+; Returns:
+;     void
+; -----------------------------------------------------------------------------
+serial_init:
+          push    rax
+          push    rdx
+
+          ; Disable interrupts: write 0 to IER (COM1+1 = 0x3F9)
+          mov     dx, COM1 + 1
+          mov     al, 0x00
+          out     dx, al
+
+          ; Set DLAB = 1 in LCR (COM1+3 = 0x3FB)
+          mov     dx, COM1 + 3
+          mov     al, DLAB_BIT
+          out     dx, al
+
+          ; Set baud rate divisor to 38400 baud (divisor = 3)
+          ; DLL (COM1+0 = 0x3F8)
+          mov     dx, COM1
+          mov     al, BAUD_DIVISOR
+          out     dx, al
+
+          ; DLM (COM1+1 = 0x3F9)
+          mov     dx, COM1 + 1
+          mov     al, 0x00
+          out     dx, al
+
+          ; Clear DLAB and set data bits (8 bits, no parity, one stop bit)
+          ; LCR (COM1+3 = 0x3FB)
+          mov     dx, COM1 + 3
+          mov     al, 0x03
+          out     dx, al
+
+          ; Enable FIFO, clear them, 14-byte threshold: FCR (COM1+2 = 0x3FA)
+          mov     dx, COM1 + 2
+          mov     al, 0xC7
+          out     dx, al
+
+          ; Enable IRQs, RTS/DSR: MCR (COM1+4 = 0x3FC)
+          mov     dx, COM1 + 4
+          mov     al, 0x0B
+          out     dx, al
+
+          ; Restore RDX and RAX registers
+          pop     rdx
+          pop     rax
+          ret
```

Comment:
It would be nice to provide the sources for hardware sensitive information when possible